src/rules/no-test-callback.ts

```diff
@@ -20,24 +25,38 @@ export default {
 
         const [, callback] = node.arguments;
 
-        if (
-          !/^(Arrow)?FunctionExpression$/.test(callback.type) ||
-          callback.params.length !== 1
-        ) {
+        if (!isFunction(callback) || callback.params.length !== 1) {
           return;
         }
 
         const [argument] = callback.params;
+
         context.report({
           node: argument,
           messageId: 'illegalTestCallback',
           fix(fixer) {
-            const sourceCode = context.getSourceCode();
             const { body } = callback;
+
+            if (!body) {
```

Comment:
could we add an istanbul ignore comment here and revisit if ts eslint change their types?

Reply:
We sure can!

The issue got traction pretty quick, and it sounds like it will be changed to not null.

I'm going to try and catch some zzzs before my 4am flight, so am happy for you to make the change and merge it. Otherwise I'll do it when I land tomorrow morning 😊

Reply:
enjoy your flight 😀